### PR TITLE
Add Button : Get Alerts for Today

### PR DESCRIPTION
**Pull request by nilpatel42**
Added Button for Getting Today Alert, Same as Frappe Notification

image

### CODE REVIEW

File: frappe_whatsapp/hooks.py [hunk 1]
```diff
@@ -136,9 +136,9 @@
 # Overriding Methods
 # ------------------------------
 #
-# override_whitelisted_methods = {
-#   "frappe.desk.doctype.event.event.get_events": "frappe_whatsapp.event.get_events"
-# }
+override_whitelisted_methods = {
```
Comment (shridarpatil) [inline frappe_whatsapp/hooks.py:139]: no need to override a method. You can remove this.

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 2]
```diff
@@ -311,3 +311,20 @@ def template():
...
     notification.send_template_message(doc)
 
+
+
+@frappe.whitelist()
+def call_trigger_notifications():
+    """Trigger notifications."""
+    try:
+        # Assuming trigger_notifications is defined elsewhere in the file
+        if 'trigger_notifications' in globals():
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:321]: The function is already defined in the same file. There is not need to check in globals

Comment (shridarpatil) [reply frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:321]: also, there is no need for a separate function you can whitelist `trigger_notifications` and use the same function.

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js [hunk 3]
```diff
@@ -107,3 +107,34 @@ frappe.ui.form.on('WhatsApp Notification', {
...
+                },
+                callback: function (response) {
+                    if (response.message && response.message.length > 0) {
+                        console.log(response.message);
+                    } else {
+                        frappe.msgprint(__('No alerts for today'));
+                    }
+                },
+                error: function (error) {
+                    console.error('Error:', error);
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js:129]: Remove console logs

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js [hunk 4]
```diff
@@ -107,3 +107,34 @@ frappe.ui.form.on('WhatsApp Notification', {
...
+    refresh: function (frm) {
+        frm.add_custom_button(__('Get Alerts for Today'), function () {
+            frappe.call({
+                method: 'frappe_whatsapp.frappe_whatsapp.doctype.whatsapp_notification.whatsapp_notification.call_trigger_notifications',
+                args: {
+                    method: 'daily' 
+                },
+                callback: function (response) {
+                    if (response.message && response.message.length > 0) {
+                        console.log(response.message);
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js:123]: Remove console logs

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 5]
```diff
@@ -296,5 +308,14 @@ def trigger_notifications(method="daily"):
...
+            # doc.name
+            # evaluate_alert(doc, alert, alert.event)
+            # frappe.db.commit()
+
+def template():
+    notification = frappe.get_doc("WhatsApp Notification", "WN-0001")
+    notification.disabled = 0
+    doc = frappe.get_doc("User", "Administrator")
+    doc.mobile_no = "+919741094468"
+
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:321]: Remove all these lines.

Comment (nilpatel42) [reply frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:321]: If I remove this line, it giving error of `\"template\" is not defined`

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 6]
```diff
@@ -261,7 +256,6 @@ def format_number(self, number):
 
         return number
 
-
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:264]: this is for flake8

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 7]
```diff
@@ -1,5 +1,3 @@
-"""Notification."""
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:1]: this is for flake8

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 8]
```diff
@@ -21,7 +19,7 @@ def validate(self):
                 filters={"dt": self.reference_doctype},
                 fields=["fieldname"]
             )
-            if not any(field.fieldname == self.field_name for field in fields): # noqa
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:24]: this is for flake8

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 9]
```diff
@@ -108,8 +105,7 @@ def send_template_message(self, doc: Document):
                 }]
 
             if self.attach_document_print:
-                # frappe.db.begin()
-                key = doc.get_document_share_key()  # noqa
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:112]: this is for flake8

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 10]
```diff
@@ -243,7 +238,7 @@ def on_trash(self):
     def after_insert(self):
         """After insert hook."""
         if self.notification_type == "Scheduler Event":
-            method = f"frappe_whatsapp.utils.trigger_whatsapp_notifications_{self.event_frequency.lower().replace(' ', '_')}" # noqa
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:246]: this is for flake8

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py [hunk 11]
```diff
@@ -286,12 +280,19 @@ def get_documents_for_today(self):
...
+    """Trigger notifications."""
+    try:
+        trigger_notifications()  
+    except Exception as e:
+        frappe.log_error(frappe.get_traceback(), "Error in call_trigger_notifications")
+        raise e
+
 def trigger_notifications(method="daily"):
     if frappe.flags.in_import or frappe.flags.in_patch:
-        # don't send notifications while syncing or patching
```
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.py:294]: do not remove comments

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js (hunk 4 above)
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js:123]: remove console logs

File: frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js (hunk 3 above)
Comment (shridarpatil) [inline frappe_whatsapp/frappe_whatsapp/doctype/whatsapp_notification/whatsapp_notification.js:129]: remove logs

**Pull request merged**